### DISCUSSION TITLE
Train a European (any other region) model

### DISCUSSION THREAD

**Discussion by krummrey** [mcguirepr89/BirdNET-Pi#771] [Ideas]
I've been running the new model for a few weeks now and I'm getting way more detections. By far most seem to be true. there are a few more false one. What I find lacking is the number of obvious calls that are not detected at all. the Great Tits (no pun intended) are out here touting all day. They show up almost perfect on the spectrogram, yet they are not reliably detected. Confidence hovers between 20-60%.

Has anyone any idea on how to train a European model? Or a model for any region? 
The supplied US model works great as it is, but it seems that there is room for improvement. 
I've trained models in the graphic domain (Pix2Pix, Informative Drawings, Stable Diffusion).  Has anyone looked into BirdNet?
We'd need a fair amount of sound files for each species we want to train on. I have the feeling that we as a community would be able to come up with those.
Then we'd need the training script and a decent GPU to train on. I've used Google Colab for that.  But more reliable options do exist for very little money.

If there is a group here that wants to join the effort we should reach out to @kahst for some guidance on how to actually get it off the ground. Or get at least some feedback on what would be required to generate a regional model that would be an improvement over what is available.

**Comment by ampsu** [comment]
One thing I noticed is that Ural owl gets mixed up with Eurasian eagle-owl... BirdNET just listens for the "uhhuus" separately and IDs the single "uuh", not the call as a whole. So I have to manually check every eagle-owl ID. Same thing with tawny and boreal owl. But well at least it detects its an owl so I could verify the exact species myself.So I guess there is a lot of room for improvements. I guess there are more species where it fails to ID correctly and mixes up with other relative. Woodpeckers also get mixed up a lot. In the end, BirdNET is just a tool that simplifies our job but does not replace it as a whole. We can not treat the results as a hard data and make conclusions simply based on them. Well, at least not yet, maybe in few years :)

**Comment by kahst** [comment]
A few thoughts from a developer perspective: We know that different models have different biases, and it's incredibly hard to predict which species will benefit and which won't. With the next model release, this could change, and other species could be distorted. I guess training a dedicated European model makes sense, but would still be a lot of work to improve performance for tits. It might be worth the effort to target threatened and rare species. We are actively working on that, and we always try to improve performance, which is very tricky. The best way of contributing to that would be to provide annotated data. BirdWeather does have a basic labeling tool, and we can access these labels to train BirdNET. Additionally, if you notice things that are off, reaching out to us and highlight those issues is also a good idea because then we can try to explore and investigate.

**Reply by ProperTeaMogul** [reply]
@kahst - I’ve been trialling the new Analyzer model here in the UK. My understanding is that this model utilises local eBird lists and species probabilities. That’s awesome and very clever! Well done to you and everyone else involved in this!

We have a lot of tawny owls nearby (we hear them most nights) but apparently they feature in less than 1.5% of eBird lists nearby. You mentioned biases in the models. Are there biases around the time of day when lists are recorded and also where lists are recorded? For instance, I suspect in the UK a lot of twitchers will go to lakes that have bird hides and record vast amounts of wildfowl however the same is not done for woodland/countryside species. In fact, according to eBird, I’m more likely to see ducks and geese in the nearby area (farm and woodland) than a robin or wren. 

In this instance, would it be best to switch back to the previous model or are there any settings that could be tweaked for better local tuning?

**Comment by krummrey** [comment]
Having my Pi and iPhone side by side, both running Birdnet, they both picks up the same birds, but at totally different quantities. (Just short tests I've done. over past few days. And there aren't many different species around here during winter) 
That might have to do with the different microphones. Or do the different location packs contain different models?

**Reply by kahst** [reply]
> Having my Pi and iPhone side by side, both running Birdnet, they both picks up the same birds, but at totally different quantities. (Just short tests I've done. over past few days. And there aren't many different species around here during winter) That might have to do with the different microphones. Or do the different location packs contain different models?

My guess is, it's different models. We have quite a few different models out there distributed across our demos and apps, so it's always a bit hard to compare.

**Comment by Skipper-is** [comment]
Yea, a European model would be great.
